src/mpl2/src/object.cpp

```diff
@@ -955,66 +977,36 @@ void HardMacro::updateDb(float pitch_x, float pitch_y, odb::dbBlock* block)
       for (odb::dbMPin* mpin : mterm->getMPins()) {
         for (odb::dbBox* box : mpin->getGeometry()) {
           odb::dbTechLayer* layer = box->getTechLayer();
-          // check the routing direction
+
           if (layer->getDirection() == odb::dbTechLayerDir::HORIZONTAL) {
-            // check the grid first
-            odb::dbTrackGrid* grid = block->findTrackGrid(layer);
-            if (grid != nullptr) {
-              std::vector<int> y_grid;
-              grid->getGridY(y_grid);
-              // use the origin as the offset and the step as the pitch
-              offset_y = dbuToMicron(static_cast<float>(y_grid[0]), dbu_);
-              pitch_y = dbuToMicron(static_cast<float>(y_grid[1] - y_grid[0]),
-                                    dbu_);
-            } else {
-              pitch_y
-                  = dbuToMicron(static_cast<float>(layer->getPitchY()), dbu_);
-              offset_y
-                  = dbuToMicron(static_cast<float>(layer->getOffsetY()), dbu_);
-            }
-            pin_width_y = dbuToMicron(static_cast<float>(box->getDY()), dbu_);
-          } else if (layer->getDirection() == odb::dbTechLayerDir::VERTICAL) {
-            odb::dbTrackGrid* grid = block->findTrackGrid(layer);
-            if (grid != nullptr) {
-              std::vector<int> x_grid;
-              grid->getGridX(x_grid);
-              // use the origin as the offset and the step as the pitch
-              offset_x = dbuToMicron(static_cast<float>(x_grid[0]), dbu_);
-              pitch_x = dbuToMicron(static_cast<float>(x_grid[1] - x_grid[0]),
-                                    dbu_);
-            } else {
-              pitch_x
-                  = dbuToMicron(static_cast<float>(layer->getPitchX()), dbu_);
-              offset_x
-                  = dbuToMicron(static_cast<float>(layer->getOffsetX()), dbu_);
-            }
-            pin_width_x = dbuToMicron(static_cast<float>(box->getDX()), dbu_);
+            computeCoordinateSpacingParameters(
+                block, layer, box, offset_y, pitch_y, pin_width_y, false);
+          }
+
+          if (layer->getDirection() == odb::dbTechLayerDir::VERTICAL) {
```

Comment:
You can merge the second `if` statement with the previous one into an `else`, so you don't need to evaluate the second check.

Reply:
There should be no pins on a layer with no direction.